db/migrations/20190919141224_create_ilk_state_history_table_and_triggers.sql

```diff
@@ -176,6 +176,52 @@ WHERE vat_init.ilk_id = ilk_time_created.ilk_id
 $$
     LANGUAGE sql;
 
+-- +goose StatementBegin
+CREATE OR REPLACE FUNCTION maker.delete_redundant_ilk_state(ilk_id INTEGER, block_number BIGINT) RETURNS api.historical_ilk_state
```

Comment:
This function is responsible for determining whether a record in `historical_ilk_state` is the same as the one before it, and deleting it if so. It's necessary because a single record in `historical_ilk_state` can have data from multiple storage tables, so we can't know whether it's safe to delete a row until we've verified that the only new data it held was from a re-orged diff.

If we can guarantee that re-orged diffs will always be deleted before new diffs are added, we can probably get rid of this function.